modules/candidate_parameters/php/module.class.inc

```diff
@@ -185,4 +188,110 @@ class Module extends \Module
         }
         return $entries;
     }
+
+    /**
+     * Return a data dictionary of data types managed by this module.
+     * DictionaryItems are grouped into categories and a module may
+     * provide 0 or more categories of dictionaryitems.
+     *
+     * @param \LORIS\LorisInstance $loris The LORIS instance whose data
+     *                                    dictionary should be retrieved.
+     *
+     * @return \LORIS\Data\Dictionary\Category[]
+     */
+    public function getDataDictionary(\LORIS\LorisInstance $loris) : iterable
+    {
+        $candscope = new Scope(Scope::CANDIDATE);
+        $sesscope  = new Scope(Scope::SESSION);
+
+        $ids = new \LORIS\Data\Dictionary\Category(
+            "Identifiers",
+            "Candidate Identifiers"
+        );
+
+        $ids = $ids->withItems(
+            [
+                new DictionaryItem(
+                    "CandID",
+                    "LORIS Candidate Identifier",
+                    $candscope,
+                    new \LORIS\Data\Types\IntegerType(999999),
+                    new Cardinality(Cardinality::UNIQUE),
+                ),
+                new DictionaryItem(
+                    "PSCID",
+                    "Project Candidate Identifier",
+                    $candscope,
+                    new \LORIS\Data\Types\StringType(255),
+                    new Cardinality(Cardinality::UNIQUE),
```

Comment:
There is no constraint on PSCID in our database schema. This should be SINGLE because there should be one value per candidate; right?

Reply:
@ridz1208 it looks like he's right.. in the SQL schema the PSCID is defined as `KEY `PSCID` (`PSCID`),` not `UNIQUE KEY`. Do you have any idea why that is? It doesn't seem right to me.

Reply:
maybe to accomodate scanners ? or non-human entities ? I think UNIQUE KEY prevents multiple values from being NULL also. is the PSCID NOT NULL in the DB ?

Reply:
Addressed in #7176